jenkins/bootstrap.py

```diff
@@ -88,6 +88,9 @@ def call(cmd, stdin=None, check=True, output=None):
                     reads.remove(fdesc)
                     continue
                 logging.warning(line[:-1])
+        if not ret[0]:
+            if proc.poll() is not None:
+                break  # process exited without closing pipes (timeout?)
```

Comment:
I didn't see a way to use the xlist for anything useful-- it *might* be able to catch when a pipe is closed, but that's not what's happening here.